kcauto.sikuli/combat.sikuli/lbas.py

```diff
@@ -146,28 +146,40 @@ def _check_and_manage_lbas_fatigue(self, fatigue, group):
                 group_fatigue['medium']
                 if group_fatigue['medium'] else fatigue['medium'])
         else:
+            Util.log_msg("LBAS group #{} has good morale.".format(group))
             lbas_group_nodes_key = 'lbas_group_{}_nodes'.format(group)
             # put LBAS group into air defense mode if it is active but no nodes
             # are assigned to it
             group_sortie_mode = (
                 'sortie'
                 if self.config.combat[lbas_group_nodes_key] else 'defense')
+            Util.log_msg("Assign LBAS group #{a} to {b} mode.".format(a=group, b=group_sortie_mode))
             self._switch_lbas_mode(group_sortie_mode)
         return fatigue
 
-    def _switch_lbas_mode(self, final_mode):
+    def _switch_lbas_mode(self, lbas_group_mode_name_target):
         """Switches the lbas group mode to the specified final mode.
 
         Args:
-            final_mode (str): the mode to switch the LBAS group to
+            lbas_group_mode_name_target (str): the mode to switch the LBAS group to
         """
         Util.rejigger_mouse(self.regions, 'top')
+        # Must be relevant to http://kancolle.wikia.com/wiki/Land_Base_Aerial_Support#Options
+        if sum([lbas_group_mode_name_target == i for i in self._lbas_group_modes_tuple]) != 1:
+            raise ValueError("No such LBAS mode: \"{}\".".format(lbas_group_mode_name_target))
         while not self.module_regions['lbas_mode_switcher'].exists(
-                'lbas_group_mode_{}.png'.format(final_mode)):
-            Util.click_preset_region(self.regions, 'lbas_mode_switch_button')
-            Util.rejigger_mouse(self.regions, 'top')
-            Util.kc_sleep(1)
-        Util.log_msg("LBAS group switched to {} mode.".format(final_mode))
+                'lbas_group_mode_{}.png'.format(lbas_group_mode_name_target), 1 + Globals.SLEEP_MODIFIER):
+            for lbas_group_mode_name_current in self._lbas_group_modes_tuple:
+                lbas_group_mode_pattern = Pattern('lbas_group_mode_{}.png'.format(
+                    lbas_group_mode_name_current)).similar(Globals.DEFAULT_SIMILARITY)
+                Util.rejigger_mouse(self.regions, 'top')
+                if self.module_regions['lbas_mode_switcher'].exists(
+                        lbas_group_mode_pattern, 1 + Globals.SLEEP_MODIFIER):
```

Comment:
You could actually just not define `lbas_group_mode_pattern` on `L173` and just use `.exists('lbas_group_mode_{}.png'.format(lbas_group_mode_name_current))`.

Also, use `Util.kc_sleep` here.

Reply:
This is linting issue.